catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveCommand.java

```diff
@@ -84,19 +88,15 @@ private Object executeRemoveFromStore() throws Exception {
         CatalogFacade catalogProvider = getCatalog();
 
         if (hasFilter()) {
-            QueryImpl query = new QueryImpl(getFilter());
-
-            query.setRequestsTotalResultsCount(true);
-            query.setPageSize(-1);
-
-            Map<String, Serializable> properties = new HashMap<>();
-            properties.put("mode", "native");
-
-            SourceResponse queryResponse = catalogProvider.query(new QueryRequestImpl(query,
-                    properties));
-
-            final List<String> idsFromFilteredQuery = queryResponse.getResults()
-                    .stream()
+            ResultIterable resultIterable = new ResultIterable(catalogFramework,
+                    new QueryRequestImpl(new QueryImpl(getFilter(),
+                            1,
+                            PAGE_SIZE,
+                            SortBy.NATURAL_ORDER,
+                            true,
+                            TimeUnit.SECONDS.toMillis(30)), false));
+
+            final List<String> idsFromFilteredQuery = resultIterable.stream()
                     .map(result -> result.getMetacard()
                             .getId())
                     .collect(Collectors.toList());
```

Comment:
Perhaps restructure to the old-fashioned foreach loop and see if looping continues indefinitely. Also reference https://github.com/codice/ddf/pull/2148#discussion-diff-129152361R134 (line 134) where the last expected result in the test mock is an emptyList (simulates when results are exhausted).